### ISSUE TITLE
single_include compilation error

### ISSUE THREAD

**Issue by viniciusjl** [vincentlaucsb/csv-parser#75]
First of all, thanks for doing this library.

When compiling files in single_include_test directory, the following compilation errors occurred:

```
$ g++ -pthread --std=c++14 -o file1 file1.cpp
In file included from my_header.hpp:2:0,
                 from file1.cpp:1:
csv.hpp:3975:28: error: enclosing class of constexpr non-static member function ‘bool csv::CSVRow::iterator::operator==(const csv::CSVRow::iterator&) const’ is not a literal type
             constexpr bool operator==(const iterator& other) const {
                            ^~~~~~~~
csv.hpp:3945:15: note: ‘csv::CSVRow::iterator’ is not literal because:
         class iterator {
               ^~~~~~~~
csv.hpp:3945:15: note:   ‘csv::CSVRow::iterator’ has a non-trivial destructor
csv.hpp:3979:28: error: enclosing class of constexpr non-static member function ‘bool csv::CSVRow::iterator::operator!=(const csv::CSVRow::iterator&) const’ is not a literal type
             constexpr bool operator!=(const iterator& other) const { return !operator==(other); }
                            ^~~~~~~~
```

To fix the errors above I've just modified the following lines:

_(the lines commented are the original ones)_

**csv.hpp**
```
        class iterator {
   ......
            /** Two iterators are equal if they point to the same field */
//|            constexpr bool operator==(const iterator& other) const {
            inline bool operator==(const iterator& other) const {
                return this->i == other.i;
            };

//|            constexpr bool operator!=(const iterator& other) const { return !operator==(other); }
            inline bool operator!=(const iterator& other) const { return !operator==(other); }
```
**file1.hpp**
```
//|int foobar(int argc, char** argv) {
int main(int argc, char** argv) {
    using namespace csv;
```
The **file2.hpp** is ok.

**Comment by viniciusjl** [comment]
BTW, the **include/internal/csv_row.hpp** file was also modified the same way as **single_include_test/csv.hpp** file above.

**Comment by vincentlaucsb** [comment]
What version of g++ are you using? Seems like a compiler bug since the Travis build using g++ succeeds with std=c++11.

https://stackoverflow.com/questions/36489369/constexpr-non-static-member-function-with-non-constexpr-constructor-gcc-clang-d

**Comment by viniciusjl** [comment]
```
$ g++ --version
g++ (Debian 6.3.0-18+deb9u1) 6.3.0 20170516
Copyright (C) 2016 Free Software Foundation, Inc.
This is free software; see the source for copying conditions.  There is NO
warranty; not even for MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.

$
```

The bug reported in the link was fixed in g++ v5.1.0.

**Comment by viniciusjl** [comment]
BTW, using std=c++11 the problem still happens.

**Comment by viniciusjl** [comment]
Well, in the bug tracker header is written that the bug was fixed in g++ v5.1.0, but reading the bug tracker comments I saw that the bug was fixed in g++ v7.2.

So, it seems a compiler bug, what g++ version was used in csv tests?

**Comment by vincentlaucsb** [comment]
https://travis-ci.org/vincentlaucsb/csv-parser/jobs/646741008

I use g++ 8. I'm thinking about applying a modification similar to yours based on g++ version since it's a simple fix, but I don't want to maintain g++ 5 compatibility forever. I'll think about it.

**Comment by viniciusjl** [comment]
I'm thinking to use clang v9 instead g++, but first I have to check the modification cost.

**Comment by vincentlaucsb** [comment]
Fixed by #179
